admin/services/data-access/insert-pupils.data.service.js

```diff
@@ -0,0 +1,10 @@
+const Pupil = require('../../models/pupil')
+
+module.exports.insertMany = async (pupils) => {
+  const savedPupils = await Pupil.insertMany(pupils, (err) => {
+    if (err) {
+      throw new Error(err)
```

Comment:
Why re-throw the error in an async function?  It will be rejected automatically in the await/async promise implementation.   Just leave it out, the caller still has to catch errors all the same.